### ISSUE TITLE
Support creating view variables in templates

### ISSUE THREAD

**Issue by mhevery** [angular/angular#507]
```
<button (click)="myDiv.hidden = !myDiv.hidden">toggle</button>
<div var-my-div>see me</div>
```

alt syntax: `<div #my-div>`

The above code allows creation of view variable `myDiv` which should than be injected and available for use as shown.

Pointers to get started:
- [View.setLocal()](https://github.com/angular/angular/blob/master/modules/core/src/compiler/view.js#L71) for setting the value.
- [ProtoView.variableBindings](https://github.com/angular/angular/blob/master/modules/core/src/compiler/view.js#L71)
- [PropertyBindingParser.process](https://github.com/angular/angular/blob/master/modules/core/src/compiler/pipeline/property_binding_parser.js#L43)

Steps needed:
1. Compiler extracts the `var` and `#` syntax from the template
1. make sure that the var information gets into `ProtoView`
1. Assign element into the variable during `ProtView` to `View` creation.

**Comment by tbosch** [comment]
The question is to what the variable should refer when there is a component
directive on the element (or a decorator directive)

**Comment by jelbourn** [comment]
```
<input #i value="works">
<div>{{i.value}}<div>
```

**Comment by mhevery** [comment]
Closed by: 6dbfe0dc2e302ed8e742d9beed3cbb6b4f014de5
